Tests/PSKoans.tests.ps1

```diff
@@ -0,0 +1,120 @@
+$here = (Split-Path -Parent $MyInvocation.MyCommand.Path) -replace 'Tests', ''
+$sut = "PSKoans.psm1"
+Import-Module "$here\$sut" -Prefix "TST"
+
+Describe -Name "Initialize-KoanDirectory Unit Testing" -Fixture {
+    Context -Name "Mocked Functions: Test-Path,Remove-Item, and Copy-Item" -Tag @("Mocking") -Fixture {
+        Mock -CommandName Test-Path -MockWith { $false }
+        Mock -CommandName Remove-Item -MockWith {}
+        Mock -CommandName Copy-Item -MockWith {}
+
+        Initialize-TSTKoanDirectory -FirstImport
+
+        It -name "Assert Test-Path mocked" -test {
+            Assert-MockCalled -CommandName Test-Path -Times 1
+        }#It
+        It -name "Assert Remove-Item mocked" -test {
+            Assert-MockCalled -CommandName Remove-Item -Times 0
+        }#It
+        It -name "Assert Copy-Item mocked" -test {
+            Assert-MockCalled -CommandName Copy-Item -Times 1
+        }#It
+    }#Context Mocked Functions
+    Context -Name "Mocked Functions: Remove-Item, and Copy-Item" -Tag @("Mocking") -Fixture {
+        $KoanFolder = $Home | Join-Path -ChildPath 'PSKoans'
+        If( Test-Path $KoanFolder){
+            Remove-Item $KoanFolder -Recurse
+        }
+        Mock -CommandName Remove-Item -MockWith {}
+        Mock -CommandName Copy-Item -MockWith {}
+
+        Initialize-TSTKoanDirectory -FirstImport
```

Comment:
Since this is not an exported command from the module... will it work to do this? Or will we need to call `Get-TSTEnlightenment -Reset` to trigger it instead?

Reply:
Yes this will work as we are importing the psm1, not the psd1. So at this point everything is exported.

Any value in later on testing what commands are exported? Maybe just a check to see that all exported commands exist in the psm1?

Reply:
Can't hurt to make sure that everything is present, absolutely!